astronomer/airflow/version_check/update_checks.py

```diff
@@ -25,18 +26,28 @@
 from airflow.utils.log.logging_mixin import LoggingMixin
 from airflow.utils.timezone import utcnow
 from airflow.www import auth
+from airflow.www.extensions.init_auth_manager import get_auth_manager
 
 try:
     from airflow.www_rbac.decorators import action_logging
 except ImportError:
     from airflow.www.decorators import action_logging
 
+if TYPE_CHECKING:
+    from airflow.auth.managers.base_auth_manager import ResourceMethod
+
+T = TypeVar("T", bound=Callable)
 
 # Code is placed in this file as the default Airflow logging config shows the
 # file name (not the logger name) so this prefixes our log messages with
 # "update_checks.py"
 
 
+def has_access(method: ResourceMethod) -> Callable[[T], T]:
+    return auth._has_access_no_details(
+        lambda: get_auth_manager().is_authorized(method=method, resource_type="UpdateAvailable"))
```

Comment:
The problem here is that you are accessing a private method. This can change without warnings

Reply:
I agree, but do we have any way we override the `FabAuthManager`?